### ISSUE TITLE
[Feature request] An option to change wallpapers based on System Theme (Dark/Light)

### ISSUE THREAD

**Issue by Username1-a** [Anthonyy232/Paperize#34]
Just would be cool if additional scheduling option was available – changing wallpapers/albums based on System Theme (with that you could have light wallpapers on light theme and dark wallpapers on dark theme)

**Comment by Anthonyy232** [comment]
Are you suggesting to have separate albums for light/dark, or having the option for the app to automatically pick a dark/light wallpaper depending on the system theme? If I'm misunderstanding, let me know! Can't promise I'll implement it as it may not be what I have in mind for the app though.

For the former: 
Consider this app, I've used it before and it works fine. It doesn't support albums but maybe that is something you can suggest to the author to implement as he is quite active. https://github.com/rajaumair7890/Darklify

For the latter: 
I might be able to implement this, but it may not be completely accurate. There may be some algorithm out there to detect this, but just off the top of my head, maybe something that classifies pixels as light or dark, and then consider the majority count. This sounds interesting and I might give it a try. May you expand on the expected behavior? Does the wallpaper switch immediately when changing device mode, or only when the wallpaper is intended to change (ex: wallpaper changes every 15 minutes, every 15 minutes it checks whether to choose a light or a dark wallpaper)?

**Comment by Anthonyy232** [comment]
Nice mockups! If you marked an album for light mode, would you be able to select another album for dark mode? Should the album and the wallpaper be changed instantly if the device changes system theme? I just want to get a full grasp on what the expected behavior is, so there is no confusion. Thanks!

**Comment by Anthonyy232** [comment]
Okay, sure, I'll see if I can make this work.

Purely visual, haven't done any logic yet, would this work?
![image](https://github.com/user-attachments/assets/aea1a68c-bb8a-42ca-b17b-276ce3497a5a)

**Comment by Anthonyy232** [comment]
What behavior would you expect if only one album is selected for either dark or light? Currently, I think if the user has only added one album, I may not even show the option enable the sync with system theme. However, if they do have more than one album but if they have selected one album to be either the dark or the light mode album, and haven't selected another album to be the other one, I'm not entirely sure what should occur. 

I think I could either:
-- automatically select another album at random to be the other one (so if user has manually selected album 0 as light mode, the app will select a random album to be the dark mode album)
-- the wallpaper changer only activates during that mode (if user selects album 0 as light mode, the wallpaper changer will only change wallpaper if the device is in light mode, regardless if it is scheduled every 1 hour)

**Comment by Anthonyy232** [comment]
Good point. I think to keep it simple, I will simply disable the wallpaper changer until the user selects a second album for the other theme. When the user has created only one album, I will gray out the the option. When the user has only selected one, I will disable the wallpaper changer until either they turn off the setting or select another album for the other theme. May need to use some tooltips or visual indicators.

**Comment by dessalines** [comment]
Most people don't actively separate their wallpapers into light and dark folders, so a much more user friendly approach, might be to use a library to detect the overall *brightness* of the upcoming image, and skip to the next one if it's not considered dark (if in dark mode)

**Comment by Anthonyy232** [comment]
I think that might be a good idea to try. It certainly simplifies things if it works well enough. I'd have to test it but I like that sound of that.

**Comment by dessalines** [comment]
There are some launchers (like niagara), that can analyze your wallpaper, and change its text color based on it. For them its impossible to manually tag wallpapers with light or dark.

**Comment by Anthonyy232** [comment]
Have changed my mind on this and will not be considering at this time, sorry!
